### PR TITLE
fix: remove ts-ignores on env.mjs

### PR DESCRIPTION
**Pull request by p13lgst**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Fix JSDoc types on `env.mjs` removing ts-ignores

💯

### CODE REVIEW

File: cli/template/base/src/env.mjs [hunk 1]
```diff
@@ -32,16 +32,21 @@ const processEnv = {
...
   const isServer = typeof window === "undefined";
 
-  const parsed = isServer
-    ? merged.safeParse(processEnv) // on server we can validate all env vars
-    : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
+  const parsed = /** @type {MergedSafeParseReturn} */ (
+    isServer
+      ? merged.safeParse(processEnv) // on server we can validate all env vars
+      : client.safeParse(processEnv) // on client we can only validate the ones that are exposed
+  );
```
Comment (juliusmarminge) [inline cli/template/base/src/env.mjs:49]: do you need the type in the middle like that? doesn't this work?

```suggestion
  /** @type {MergedSafeParseReturn} */
  const parsed = isServer
    ? merged.safeParse(processEnv) // on server we can validate all env vars
    : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
```

**Review by juliusmarminge — COMMENTED** [review]
Interesting. I like it!

We need to do it for all the other files too though

File: cli/template/base/src/env.mjs (hunk 1 above)
Comment (p13lgst) [reply cli/template/base/src/env.mjs:49]: It doesn't work like that, we need this syntax for the type casting: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#casts

Comment (juliusmarminge) [reply cli/template/base/src/env.mjs:49]: I see

Comment (p13lgst) [reply cli/template/base/src/env.mjs:48]: Screenshot from 2023-02-09 17-43-15

The type casting works kinda like the `as` keyword in TS

**Comment by bai** [comment]
This is badass ✨ Thanks for contributing!

**Review by c-ehrlich — APPROVED** [review]
looks good to me how it is now, but would like a 2nd approval from @juliusmarminge before merging

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**